conduit/plugins/processors/filterprocessor/filter_processor.go

```diff
@@ -132,6 +132,32 @@ func (a *FilterProcessor) Process(input data.BlockData) (data.BlockData, error)
 			return data.BlockData{}, err
 		}
 	}
+	if !a.cfg.OmitGroupTransactions {
+		gtxns := groupTxns(&input.Payset)
+		matchedTxns := payset
+		payset = make([]sdk.SignedTxnInBlock, 0)
+		included := make(map[sdk.Digest]bool)
+		for _, match := range matchedTxns {
+			if match.Txn.Group != (sdk.Digest{}) {
+				if _, ok := included[match.Txn.Group]; !ok {
+					// txn has a groupID, get all the txns in the same group
+					payset = append(payset, gtxns[match.Txn.Group]...)
+					included[match.Txn.Group] = true
+				}
+			} else {
+				payset = append(payset, match)
+			}
+		}
+	}
```

Comment:
Got a little carried away:
```diff
// SearchAndFilter searches through the block data and applies the operation to the results
func (f Filter) SearchAndFilter(payset []sdk.SignedTxnInBlock) ([]sdk.SignedTxnInBlock, error) {
	var result []sdk.SignedTxnInBlock
+	var firstGroupIDX int
-	for _, txn := range payset {
+	for i := 0; i < len(payset); i++ {
+		if !bytes.Equal(payset[firstGroupIDX].group, payset[i].group) {
+			firstGroupIDX = i
+		}
		match, err := f.matches(&txn.SignedTxnWithAD)
		if err != nil {
			return nil, err
		}
		if match {
-			result = append(result, txn)
+			if !OmitGroup {
+				iterate from firstGroupIDX until the end of the payset, or the group changes.
+				update firstGroupIDX and i so that you skip over transactions which aren't needed. 
+			} else {
+				result = append(result, txn)
+			}
		}
	}

	return result, nil
}
```

Reply:
I see, I didn't realize transactions in the same group are adjacent.